linera-execution/src/runtime.rs

```diff
@@ -402,10 +402,13 @@ impl SyncRuntimeInternal<UserContractInstance> {
             _ => None,
         };
         let authenticated_caller_id = authenticated.then_some(caller_id);
-        let callee_context = CalleeContext {
+        let callee_context = OperationContext {
             chain_id: self.chain_id,
             authenticated_signer,
             authenticated_caller_id,
+            height: self.height,
+            index: None,
```

Comment:
We could still make this non-optional, and always use the current transaction index, i.e. the index of the message or operation in the current block, which directly or indirectly caused this operation.

Reply:
Yeah, I wasn't exactly sure how to merge the `OperationContext` and the `CalleeContext`, so there are a few tweaks that we could make. We can discuss it in a call and I can open a separate PR with the changes :+1: